### DISCUSSION TITLE
Java: Saving a modified map in png/gif format with no gui

### DISCUSSION THREAD

**Discussion by aarchett** [mapsforge/mapsforge#1330] [Q&A] [answered]
Hi all,
I'm new to using Maps libraries and using Java (not Android) I'm trying to do the following:
1. load a downloaded map file
2. put some markers and draw a few lines between those markers
3. save the map with my markers and lines into a png/gif file.

I have been able to do all of this using JFrame and awt containers (MapView) but I would like to get the same result without showing any gui. 
The use case is the following: the final user wants to receive via email a pdf with the modified map image and my java code is running on a docker container on a linux server.

Can you suggest if this is feasible with mapsforge and how?

Thanks,
Antonella

**Comment by devemux86** [comment, marked as answer]
Without map view, only map tiles may be saved, see the [sample](https://github.com/mapsforge/mapsforge/blob/master/mapsforge-samples-awt/src/main/java/org/mapsforge/samples/awt/SaveTiles.java).

The result may not be the best, as label / symbols rendering optimizations are only possible in map view / label layer.

**Reply by aarchett** [reply]
thanks for your reply: without it I would have tried for a few more days without realizing it is not feasible.
